ui/src/app/components/direct-connection/direct-connection.component.ts

```diff
@@ -51,10 +55,16 @@ export class DirectConnectionComponent implements OnInit {
     private data: DataService,
     private loader: LoaderService,
     private snackbarService: SnackbarService,
-    private clickEvent: ClickEventService
+    private clickEvent: ClickEventService,
+    private dialog: MatDialog,
   ) {}
 
   ngOnInit(): void {
+    this.data.isOffline.subscribe({
```

Comment:
This is non synchronous call, which I feel can call issues as we need to open the box synchronously

Reply:
the  call will require because based on that, we are showing the popup and restricting user to enter